test/unit/stateTransition/validation/validateIdentityExistenceAndTypeFactory.spec.js

```diff
@@ -0,0 +1,77 @@
+const Identity = require('../../../../lib/identity/Identity');
+const IdentityPublicKey = require('../../../../lib/identity/IdentityPublicKey');
+
+const validateIdentityExistenceAndTypeFactory = require('../../../../lib/stateTransition/validation/validateIdentityExistenceAndTypeFactory');
+
+const createDataProviderMock = require('../../../../lib/test/mocks/createDataProviderMock');
+
+const { expectValidationError } = require('../../../../lib/test/expect/expectError');
+
+const ValidationResult = require('../../../../lib/validation/ValidationResult');
+
+const IdentityNotFoundError = require('../../../../lib/errors/IdentityNotFoundError');
+const UnexpectedIdentityTypeError = require('../../../../lib/errors/UnexpectedIdentityTypeError');
+
+describe('validateIdentityExistenceAndType', () => {
+  let validateIdentityExistenceAndType;
+  let dataProviderMock;
+  let userId;
+  let rawIdentityUser;
+
+  beforeEach(function beforeEach() {
+    dataProviderMock = createDataProviderMock(this.sinonSandbox);
+
+    validateIdentityExistenceAndType = validateIdentityExistenceAndTypeFactory(
+      dataProviderMock,
+    );
+
+    userId = 'iTYF+bWBA4MYRURcsBpBkgfwiqV7sYVnTDPR4uQ/KLU=';
```

Comment:
Need to change to base58, as we agreed that this is the best way to make browser links in explorers work